pkg/k8s/deployments/crud_test.go

```diff
@@ -28,28 +28,162 @@ import (
 )
 
 func TestGet(t *testing.T) {
-	ctx := context.Background()
-	deployment := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "fake",
-			Namespace: "test",
+	tests := []struct {
```

Comment:
I've added test cases that I thought were meaningful. Because I might not have the full context yet, I appreciate if there is feedback on this, and I am happy to adjust 🙏

Reply:
I think those scenarios are perfect, @jLopezbarb can you check them too?

Reply:
I think they are the ones to have. We could add also another scenario (very weird scenario): `Search by Label: Already in dev`. For this use case the deployment that is returned by the client should have `dev.okteto.com/clone` set.
This use case is a very weird scenario where the user creates the environment with the `autocreate` and then tries to connect to that dev which is not supported and we should return a `not found` error

Reply:
I think the tests are perfect!!! Review my other comment in case that test is needed but other than that LGTM